Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebExtensionAPITest.mm

```diff
@@ -218,6 +218,42 @@
     EXPECT_NS_EQUAL(firstMessage, @{ @"key": @"One" });
 }
 
+TEST(WKWebExtensionAPITest, AddTestThatPasses)
+{
+    NSString *testName = @"browser.test.addTest call that rejects";
+    auto *backgroundScript = Util::constructScript(@[
+        @"browser.test.assertResolves(browser.test.addTest(async () => {",
+        @"  browser.test.assertTrue(true)",
+        [NSString stringWithFormat:@"}, '%@'))", testName],
+        @"  .then(() => browser.test.notifyPass())",
+        @"  .catch(() => browser.test.notifyFail())"
+    ]);
+
+    auto manager = Util::loadExtension(manifest, @{ @"background.js": backgroundScript });
+
+    [manager run];
+    EXPECT_NS_EQUAL(manager.get().testsAdded, @[ testName ]);
+    EXPECT_NS_EQUAL(manager.get().testsStarted, @[ testName ]);
```

Comment:
```suggestion
    [manager run];

    EXPECT_NS_EQUAL(manager.get().testsAdded, @[ testName ]);
    EXPECT_NS_EQUAL(manager.get().testsStarted, @[ testName ]);
```

This is a little odd, since the tests have ended by the time we get here.

With my other suggestion, this turns into this:

```suggestion
    [manager run];

    EXPECT_NS_EQUAL(manager.get().testsAdded, @[ testName ]);
    EXPECT_NS_EQUAL(manager.get().testsStarted, @[ testName ]);
    EXPECT_NS_EQUAL(manager.get().testResults, @{ testName: @YES });
```